### PR TITLE
add command to rm merged branches

### LINKED ISSUE #14: RFE: New command: purge local branches which were merged already
I have projects where I have dozens of branches which could be cleaned since the respective PRs were merged already.

TODO:
* [ ] remove a branch if the PR was merged already #37
* [ ] by default, request confirmation from user that branches [...] will be removed #37
* [ ] add --force option to bypass
* [ ] add --remote option to delete on remote as well (gracefully)

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes #14

+ list-branches: add an option to pick a "merged with" branch

### CODE REVIEW

**Comment by TomasTomecek** [comment]
rebased

**Comment by dhodovsk** [comment]
New option for list-branches lists all local branches for me :( 

```
[dhodovsk@shiny packit-tmp]$ upsint list-branches --merged-with master
╒═════════════════╤═══════════════╤═════════════════════╤══════════════╤════════╕
│ version-compare │               │ 2019-03-29 15:23:10 │              │        │
├─────────────────┼───────────────┼─────────────────────┼──────────────┼────────┤
│ flask           │               │ 2019-03-26 11:52:27 │              │        │
├─────────────────┼───────────────┼─────────────────────┼──────────────┼────────┤
│ master          │ origin/master │ 2019-03-04 12:10:05 │ [behind 287] │ merged │
╘═════════════════╧═══════════════╧═════════════════════╧══════════════╧════════╛
[dhodovsk@shiny packit-tmp]$ git branch cheers
[dhodovsk@shiny packit-tmp]$ upsint list-branches --merged-with master
╒═════════════════╤═══════════════╤═════════════════════╤══════════════╤════════╕
│ cheers          │               │ 2019-03-29 15:23:10 │              │        │
├─────────────────┼───────────────┼─────────────────────┼──────────────┼────────┤
│ version-compare │               │ 2019-03-29 15:23:10 │              │        │
├─────────────────┼───────────────┼─────────────────────┼──────────────┼────────┤
│ flask           │               │ 2019-03-26 11:52:27 │              │        │
├─────────────────┼───────────────┼─────────────────────┼──────────────┼────────┤
│ master          │ origin/master │ 2019-03-04 12:10:05 │ [behind 287] │ merged │
╘═════════════════╧═══════════════╧═════════════════════╧══════════════╧════════╛

```

File: upsint/cli.py [hunk 1]
```diff
@@ -192,13 +196,48 @@ def update_labels(source_repo, service, source_service, destination):
...
+        if branch_dict["merged"] == "merged":
+            to_remove.append(branch_name)
+    if not to_remove:
+        print("Nothing to remove.")
+        return
+    print("Shall we remove these local branches?")
+    for b in to_remove:
+        print(f"* {b}")
+    inp = input("Y/N? ")
+    if inp in ("y", "Y", "yolo"):
```
Comment (dhodovsk) [inline upsint/cli.py]: It is angry with me when I answer (Y or N):

```
[dhodovsk@shiny packit-tmp]$ upsint remove-merged-branches
Shall we remove these local branches?
* version-compare
* flask
Y/N? Y
Traceback (most recent call last):
  File "/usr/local/bin/upsint", line 11, in <module>
    load_entry_point('upsint==0.1.dev72+g42ebe9b', 'console_scripts', 'upsint')()
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/upsint/cli.py", line 224, in remove_merged_branches
    import ipdb; ipdb.set_trace()
ModuleNotFoundError: No module named 'ipdb'
```

File: upsint/cli.py [hunk 2]
```diff
@@ -92,12 +92,16 @@ def list_prs(service, repo):
                     "remote tracking branch, date, divergence status and "
                     "whether the branch was merged to master."
                )
-def list_branches():
+@click.option(
+    '--merged-with', type=click.STRING, default="master",
```
Comment (dhodovsk) [inline upsint/cli.py]: upsint ignores the option in my repo :(
```
[dhodovsk@shiny packit-tmp]$ upsint list-branches --merged-with master
╒═════════════════╤═══════════════╤═════════════════════╤══╤════════╕
│ master          │ origin/master │ 2019-04-01 11:47:52 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ version-compare │               │ 2019-03-29 15:23:10 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ flask           │               │ 2019-03-26 11:52:27 │  │ merged │
╘═════════════════╧═══════════════╧═════════════════════╧══╧════════╛
[dhodovsk@shiny packit-tmp]$ git checkout -b cheers
Switched to a new branch 'cheers'
[dhodovsk@shiny packit-tmp]$ upsint list-branches --merged-with master
╒═════════════════╤═══════════════╤═════════════════════╤══╤════════╕
│ cheers          │               │ 2019-04-01 11:47:52 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ master          │ origin/master │ 2019-04-01 11:47:52 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ version-compare │               │ 2019-03-29 15:23:10 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ flask           │               │ 2019-03-26 11:52:27 │  │ merged │
╘═════════════════╧═══════════════╧═════════════════════╧══╧════════╛
[dhodovsk@shiny packit-tmp]$ touch this
[dhodovsk@shiny packit-tmp]$ git add this
[dhodovsk@shiny packit-tmp]$ git commit -m "Very important commit"
[cheers b915bc3] Very important commit
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 this
[dhodovsk@shiny packit-tmp]$ upsint list-branches --merged-with master
╒═════════════════╤═══════════════╤═════════════════════╤══╤════════╕
│ cheers          │               │ 2019-04-01 12:23:02 │  │        │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ master          │ origin/master │ 2019-04-01 11:47:52 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ version-compare │               │ 2019-03-29 15:23:10 │  │ merged │
├─────────────────┼───────────────┼─────────────────────┼──┼────────┤
│ flask           │               │ 2019-03-26 11:52:27 │  │ merged │
╘═════════════════╧═══════════════╧═════════════════════╧══╧════════╛
```

But it lists correct branches when removing :)
```
[dhodovsk@shiny packit-tmp]$ upsint remove-merged-branches
Shall we remove these local branches?
* version-compare
* flask
Y/N? 
```

File: upsint/cli.py (hunk 1 above)
Comment (TomasTomecek) [reply upsint/cli.py]: Yeah, I forgot to remove debugger -_-

Comment (TomasTomecek) [reply upsint/cli.py]: Oh, I pushed it last week, please fetch latest commits.

File: upsint/cli.py (hunk 2 above)
Comment (TomasTomecek) [reply upsint/cli.py]: The series of commands you posted seem legit to me: what is your expectation?

Comment (dhodovsk) [reply upsint/cli.py]: I expected that when adding `--merged-with master` it won't include branch `cheers` in the list.

Comment (TomasTomecek) [reply upsint/cli.py]: oh, now I get it: it does not work as filter, it works as a query (for the last column) - we can turn it into a filter

Comment (dhodovsk) [reply upsint/cli.py]: No need, that was just nitpicking, Now I get it.

**Comment by TomasTomecek** [comment]
@dhodovsk thank you for the review! I'll turn your suggestion into an issue

**Pull request merged**